src/NumericalAlgorithms/Spectral/Clenshaw.hpp

```diff
@@ -0,0 +1,44 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+#pragma once
+
+#include <cmath>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace Spectral {
+/// Clenshaw's algorithm for evaluating linear combinations of
+/// basis functions obeying a given three term recurrence relation.
+/// See numerical recipes 3rd edition 5.4.2 for details.
+/// `alpha` and `beta` define the recursion
+/// \f$ f_n(x) = alpha f_{n-1}(x) + beta f_{n-2}(x) \f$
+/// `f0_of_x` and `f1_of_x` are \f$ f_0(x)\f$ and \f$f_1(x)\f$ respectively
+template <typename CoefficientDataType, typename DataType>
+DataType evaluate_clenshaw(const std::vector<CoefficientDataType>& coefficients,
+                           DataType alpha, DataType beta, DataType f0_of_x,
+                           DataType f1_of_x) {
```

Comment:
Const-reference for all of these unless I'm missing them being modified by the function.